src/librustdoc/html/render/mod.rs

```diff
@@ -170,6 +170,10 @@ crate struct SharedContext<'tcx> {
 }
 
 impl<'tcx> Context<'tcx> {
+    fn depth(&self) -> usize {
+        self.current.len()
```

Comment:
I'm confused - did `current` already exist? How is it kept in sync with `CURRENT_DEPTH`?

Reply:
`CURRENT_DEPTH` is set to `self.current.len()` in `render_item()`. `self.current` should always be the same when rendering a particular item since it represents the namespace hierarchy.

Reply:
I can combine this PR with #82815 so that we don't have to worry about keeping `CURRENT_DEPTH` in sync with `depth()`.